### DISCUSSION TITLE
Pop-up window error appears when ending experiment?

### DISCUSSION THREAD

**Discussion by jubejubess** [jspsych/jsPsych#1843] [Q&A]
Hello,

For my experiment, the participant ends the study by clicking a "CLOSE" button at the end. However, participants only see a blank screen with a completed progress bar at the top once they reach the end and when they try to close the browser, a pop-up window appears saying if they close the browser there might be unsaved changes.

I tried adding the following code to the screen with the close button, but it still gets stuck on a blank page with the progress bar and they still get the pop-up window with the same message. 

`on_finish: function(data){
      jsPsych.endExperiment();
      }`

Is there another way of automatically ending the experiment? Thanks!

**Comment by becky-gilbert** [comment]
Hi @jubejubess, are you using cognition.run to host your experiment?
